From 20f211dc5c76b00ae4665b6814abf8cc8914d843 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Thu, 23 Mar 2023 16:33:43 +0100
Subject: [PATCH] GOOD-ENH: Enable installation of resources

This enables the installation of `qSlicer.qrc` and the `Logos`
directory. In addition this exposes new CMake variables for resources.
---
 CMake/SlicerDirectories.cmake          |  3 +++
 CMakeLists.txt                         | 25 ++++++++++++++++++-------
 Templates/SlicerInstallConfig.cmake.in |  2 ++
 3 files changed, 23 insertions(+), 7 deletions(-)

diff --git a/CMake/SlicerDirectories.cmake b/CMake/SlicerDirectories.cmake
index cbed306a26..082150a3ec 100644
--- a/CMake/SlicerDirectories.cmake
+++ b/CMake/SlicerDirectories.cmake
@@ -193,3 +193,6 @@ set(Slicer_INSTALL_THIRDPARTY_SHARE_DIR "${Slicer_INSTALL_ROOT}${Slicer_THIRDPAR
 
 set(Slicer_UTILITY_SCRIPTS_DIR "${CMAKE_SOURCE_DIR}/Utilities/Scripts")
 set(Slicer_INSTALL_UTILITY_SCRIPTS_DIR "${Slicer_INSTALL_LIBEXEC_DIR}/Utilities/Scripts")
+
+set(Slicer_RESOURCES_DIR "${CMAKE_SOURCE_DIR}/Resources")
+set(Slicer_INSTALL_RESOURCES_DIR "${Slicer_INSTALL_SHARE_DIR}/Resources")
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7a989e5433..3ceb675cc1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,13 +126,6 @@ endif()
 include(SlicerInitializeBuildType)
 include(SlicerInitializeReleaseType)
 
-#-----------------------------------------------------------------------------
-# Set the header template which defines custom export/import macros
-# for shared libraries
-#-----------------------------------------------------------------------------
-set(Slicer_EXPORT_HEADER_TEMPLATE "${Slicer_SOURCE_DIR}/Templates/qSlicerExport.h.in")
-set(Slicer_LOGOS_RESOURCE "${Slicer_SOURCE_DIR}/Resources/qSlicer.qrc")
-
 #-----------------------------------------------------------------------------
 # Platform checks
 #-----------------------------------------------------------------------------
@@ -508,6 +501,14 @@ include(SlicerBlockCXXRequiredFlags)
 #-----------------------------------------------------------------------------
 include(SlicerDirectories)
 
+#-----------------------------------------------------------------------------
+# Set the header template which defines custom export/import macros
+# for shared libraries
+#-----------------------------------------------------------------------------
+set(Slicer_EXPORT_HEADER_TEMPLATE "${Slicer_SOURCE_DIR}/Templates/qSlicerExport.h.in")
+set(Slicer_LOGOS_RESOURCE "${Slicer_SOURCE_DIR}/Resources/qSlicer.qrc")
+set(Slicer_LOGOS_INSTALL_RESOURCE "${Slicer_INSTALL_SHARE_DIR}/Resources/qSlicer.qrc")
+
 # SlicerExecutionModel output directories
 if(Slicer_SUPERBUILD)
   set(_sem_output_dir ${CMAKE_BINARY_DIR}/${Slicer_BINARY_INNER_SUBDIR})
@@ -1282,6 +1283,16 @@ endif()
 install(PROGRAMS ${Slicer_QRCC_SCRIPT}
   DESTINATION ${Slicer_INSTALL_UTILITY_SCRIPTS_DIR})
 
+#-----------------------------------------------------------------------------
+# Install Resources
+#-----------------------------------------------------------------------------
+if(NOT Slicer_INSTALL_NO_DEVELOPMENT)
+  install(FILES ${Slicer_LOGOS_RESOURCE}
+    DESTINATION ${Slicer_INSTALL_RESOURCES_DIR})
+  install(DIRECTORY ${Slicer_RESOURCES_DIR}/Logos
+    DESTINATION ${Slicer_INSTALL_RESOURCES_DIR})
+endif()
+
 #-----------------------------------------------------------------------------
 # Create target to download data from the <Slicer_ExternalData_DATA_MANAGEMENT_TARGET> group.
 # This must come after all tests have been added that reference the group, so we put it last.
diff --git a/Templates/SlicerInstallConfig.cmake.in b/Templates/SlicerInstallConfig.cmake.in
index 67cf9a556c..e9d29627fd 100644
--- a/Templates/SlicerInstallConfig.cmake.in
+++ b/Templates/SlicerInstallConfig.cmake.in
@@ -167,3 +167,5 @@ set(Slicer_Libs_VTK_WRAPPED_LIBRARIES "@Slicer_Libs_VTK_WRAPPED_LIBRARIES@")
 set(Slicer_QRCC_SCRIPT "@Slicer_INSTALL_UTILITY_SCRIPTS_DIR@/qrcc.py")
 
 set(Slicer_EXTENSION_GENERATE_CONFIG "@Slicer_EXTENSION_INSTALL_GENERATE_CONFIG@")
+
+set(Slicer_LOGOS_RESOURCE "@Slicer_LOGOS_INSTALL_RESOURCE@")
-- 
2.39.2

